services/movie.py

```diff
@@ -0,0 +1,46 @@
+from typing import Optional
+
+from django.db.models import Q
+
+from django.db.models.query import QuerySet
+
+from db.models import Movie, Genre, Actor
+
+
+def get_movies(
+        genres_ids: Optional[list] = None,
+        actors_ids: Optional[list] = None
+) -> QuerySet[Movie]:
+    if not genres_ids and not actors_ids:
+        return Movie.objects.all()
+    elif genres_ids and actors_ids:
+        return Movie.objects.filter(
+            Q(genres__in=genres_ids) & Q(actors__in=actors_ids)
+        )
+    elif genres_ids and not actors_ids:
+        return Movie.objects.filter(genres__in=genres_ids)
+    elif not genres_ids and actors_ids:
+        return Movie.objects.filter(actors__in=actors_ids)
+

```

Comment:
You can use only one return and couple of "if" statements here. You can create one queryset for ocasion when there are no optional arguments. and then filter it if the argument is present. and finally return this queryset.

Reply:
Yep, nice idea! (y) Thanks )